src/sentry/buffer/redis.py

```diff
@@ -429,10 +456,8 @@ def _process(
 
     def _process_single_incr(self, key: str) -> None:
         client = get_cluster_routing_client(self.cluster, self.is_redis_cluster)
-        lock_key = self._make_lock_key(key)
-        # prevent a stampede due to the way we use celery etas + duplicate
-        # tasks
-        if not client.set(lock_key, "1", nx=True, ex=10):
+        lock_key = self._lock_key(client, key, ex=10)
+        if not lock_key:
             metrics.incr("buffer.revoked", tags={"reason": "locked"}, skip_internal=False)
             logger.debug("buffer.revoked.locked", extra={"redis_key": key})
             return
```

Comment:
is there more we can DRY up between `_process_single_incr` `incr` and `process_batch`? For example, i'm not sure if we can simplify the pipe stuff on 467 -> 474 or not.

Reply:
I changed `get_redis_connection` a teeny bit to accept a transaction boolean since that was the only difference I saw in a couple other usages of almost the same code and cleaned that up.